### PR TITLE
JENKINS-62788 Remove custom f:checkbox fork

### PR DESCRIPTION
**Pull request by timja**
https://issues.jenkins-ci.org/browse/JENKINS-62788

Admin user:
image

User with Configure:

image

cc @fqueiruga @Vlatombe @jglick 

The baseline in this plugin is new enough that the required code is in core already 😄

### CODE REVIEW

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesFolderProperty/config.jelly [hunk 1]
```diff
@@ -14,7 +14,8 @@
 					<j:otherwise>
 						<j:forEach var="permission" items="${effectivePermissions}">
 							<f:entry field="${permission.formName}">
-								<kube:checkbox title="${permission.name}" checked="${permission.granted}" readonly="${permission.readonly}"/>
+								<j:set var="readOnlyMode" value="${permission.readonly}"/>
+								<f:checkbox title="${permission.name}" checked="${permission.granted}" />
```
Comment (jglick) [inline src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesFolderProperty/config.jelly:18]:
```suggestion
								<f:checkbox title="${permission.name}" checked="${permission.granted}"  readonly="${permission.readonly}"/>
```

does not work?

**Comment by jglick** [comment]
https://github.com/jenkinsci/kubernetes-plugin/pull/282#discussion_r171800979 for context. But https://github.com/jenkinsci/jenkins/pull/2073 predates this. What am I missing?

File: src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesFolderProperty/config.jelly (hunk 1 above)
Comment (timja) [reply src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesFolderProperty/config.jelly:18]: That only prevents clicking through JavaScript it doesn’t set the standard DOM property for it and it doesn’t change the UI to show they can’t click it.

At least from reading the code I haven’t tried it...

Comment (jglick) [reply src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesFolderProperty/config.jelly:18]: Looks like I was confusing `readonly`, old, with `readOnlyMode` introduced in https://github.com/jenkinsci/jenkins/pull/4506. It seems the latter is not an attribute but an (undocumented?) dynamically-scoped variable.

Comment (timja) [reply src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesFolderProperty/config.jelly:18]: It’s documented on Jenkins.io https://www.jenkins.io/doc/developer/views/read-only/#enabling-read-only-view-support

Comment (jglick) [reply src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesFolderProperty/config.jelly:18]: I see. Can you link to that from the attribute docs for `readonly` on affected controls please?

Comment (timja) [reply src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesFolderProperty/config.jelly:18]: it's not an attribute though, would you expect it as part of the top level description?

Comment (jglick) [reply src/main/resources/org/csanchez/jenkins/plugins/kubernetes/KubernetesFolderProperty/config.jelly:18]: I am just guessing someone trying to figure out how to make a checkbox read-only is going to look at the documentation for the attribute named `readonly`, so put it there.

**Comment by Vlatombe** [comment]
> [#282 (comment)](https://github.com/jenkinsci/kubernetes-plugin/pull/282#discussion_r171800979) for context. But [jenkinsci/jenkins#2073](https://github.com/jenkinsci/jenkins/pull/2073) predates this. What am I missing?

https://github.com/jenkinsci/kubernetes-plugin/pull/282#discussion_r171800979

Looks fine with the new code though.

**Pull request merged**